src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -31,13 +31,27 @@
  */
 final class EntityValueResolver implements ValueResolverInterface
 {
+    /** @var array<class-string, class-string> */
+    private array $typeAliases = [];
+
     public function __construct(
         private ManagerRegistry $registry,
         private ?ExpressionLanguage $expressionLanguage = null,
         private MapEntity $defaults = new MapEntity(),
     ) {
     }
 
+    /**
+     * Adds an original class name to resolve to a target class name.
+     *
+     * @param class-string $original
+     * @param class-string $target
+     */
+    public function addTypeAlias(string $original, string $target): void
```

Comment:
let's move the map as a constructor argument (as advised in https://github.com/doctrine/DoctrineBundle/pull/1790/files#r1562434473)